consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/tests/RoundChangeTest.java

```diff
@@ -0,0 +1,319 @@
+
+    final SignedData<PreparePayload> prepare =
+        roles.getNonProposingPeer(0).injectPrepare(roundId, blockToPropose.getHash());
+    roles.getAllPeers().forEach(ValidatorPeer::clearReceivedMessages);
+
+    final SignedData<RoundChangePayload> expectedTxRoundChange =
+        localNodeMessageFactory.createSignedRoundChangePayload(
+            targetRound,
+            Optional.of(
+                new PreparedCertificate(
+                    proposal, Lists.newArrayList(localPrepareMessage, prepare))));
+
+    context.getController().handleRoundExpiry(new RoundExpiry(roundId));
+    assertPeersReceivedExactly(roles.getAllPeers(), expectedTxRoundChange);
+  }
+
+  @Test
+  public void whenSufficientRoundChangeMessagesAreReceivedForNewRoundLocalNodeCreatesNewRoundMsg() {
+    // Note: Round-3 is the next round for which the local node is Proposer
+    final ConsensusRoundIdentifier targetRound = new ConsensusRoundIdentifier(1, 3);
+    final Block locallyProposedBlock =
+        context.createBlockForProposal(targetRound.getRoundNumber(), blockTimeStamp);
+
+    final SignedData<RoundChangePayload> rc1 =
+        roles.getNonProposingPeer(0).injectRoundChange(targetRound, empty());
+    final SignedData<RoundChangePayload> rc2 =
+        roles.getNonProposingPeer(1).injectRoundChange(targetRound, empty());
+    final SignedData<RoundChangePayload> rc3 =
+        roles.getProposer().injectRoundChange(targetRound, empty());
+
+    final SignedData<NewRoundPayload> expectedNewRound =
+        localNodeMessageFactory.createSignedNewRoundPayload(
+            targetRound,
+            new RoundChangeCertificate(Lists.newArrayList(rc1, rc2, rc3)),
+            localNodeMessageFactory.createSignedProposalPayload(targetRound, locallyProposedBlock));
+
+    assertPeersReceivedExactly(roles.getAllPeers(), expectedNewRound);
+  }
+
+  @Test
+  public void newRoundMessageContainsBlockOnWhichPeerPrepared() {
+    final long ARBITRARY_BLOCKTIME = 1500;
+
+    final ConsensusRoundIdentifier earlierPreparedRound = new ConsensusRoundIdentifier(1, 1);
+    final ConsensusRoundIdentifier bestPreparedRound = new ConsensusRoundIdentifier(1, 2);
+    final ConsensusRoundIdentifier targetRound = new ConsensusRoundIdentifier(1, 3);
+
+    final Block earlierPreparedBlock = context.createBlockForProposal(1, ARBITRARY_BLOCKTIME / 2);
+    final Block bestPreparedBlock = context.createBlockForProposal(2, ARBITRARY_BLOCKTIME);
+
+    final SignedData<RoundChangePayload> rc1 =
+        roles.getNonProposingPeer(0).injectRoundChange(targetRound, empty());
+
+    // Create a roundChange with a PreparedCertificate from an earlier Round (should not be used
+    final SignedData<RoundChangePayload> rc2 =
+        roles
+            .getNonProposingPeers()
+            .get(1)
+            .injectRoundChange(
+                targetRound,
+                Optional.of(
+                    new PreparedCertificate(
+                        roles
```

Comment:
be create some small helper functions within the test for the prepare and proposal payload, think that would make this test a bit more readable

Reply:
agreed - done.